src/pages/NewReportWorkspaceSelectionPage.tsx

```diff
@@ -0,0 +1,132 @@
+
+type WorkspaceListItem = {
+    text: string;
+    policyID?: string;
+    isPolicyAdmin?: boolean;
+} & ListItem;
+
+function NewReportWorkspaceSelectionPage() {
+    const {isOffline} = useNetwork();
+    const styles = useThemeStyles();
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const {translate} = useLocalize();
+
+    const [policies, fetchStatus] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+    const [isLoadingApp] = useOnyx(ONYXKEYS.IS_LOADING_APP);
+    const shouldShowLoadingIndicator = isLoadingApp && !isOffline;
+
+    const selectPolicy = useCallback(
+        (policyID?: string) => {
+            if (!policyID) {
+                return;
+            }
+            const createdReportID = createNewReport(policyID, currentUserPersonalDetails);
+            Navigation.navigate(ROUTES.SEARCH_MONEY_REQUEST_REPORT.getRoute({reportID: createdReportID, backTo: Navigation.getActiveRoute()}), {forceReplace: true});
+        },
+        [currentUserPersonalDetails],
+    );
+
+    const usersWorkspaces = useMemo<WorkspaceListItem[]>(() => {
+        if (!policies || isEmptyObject(policies)) {
+            return [];
+        }
+
+        return Object.values(policies)
+            .filter((policy) => shouldShowPolicy(policy, !!isOffline, currentUserPersonalDetails?.login) && !policy?.isJoinRequestPending && policy?.isPolicyExpenseChatEnabled)
+            .map((policy) => ({
+                text: policy?.name ?? '',
+                policyID: policy?.id,
+                icons: [
+                    {
+                        source: policy?.avatarURL ? policy.avatarURL : getDefaultWorkspaceAvatar(policy?.name),
+                        fallbackIcon: Expensicons.FallbackWorkspaceAvatar,
+                        name: policy?.name,
+                        type: CONST.ICON_TYPE_WORKSPACE,
+                        id: policy?.id,
+                    },
+                ],
+                keyForList: policy?.id,
+                isPolicyAdmin: isPolicyAdmin(policy),
+                shouldSyncFocus: true,
+            }));
+    }, [policies, isOffline, currentUserPersonalDetails?.login]);
+
+    const filteredAndSortedUserWorkspaces = useMemo<WorkspaceListItem[]>(
+        () => usersWorkspaces.filter((policy) => policy.text?.toLowerCase().includes(debouncedSearchTerm?.toLowerCase() ?? '')),
+        [debouncedSearchTerm, usersWorkspaces],
+    );
+
+    const sections = useMemo(() => {
+        const options: Array<SectionListDataType<WorkspaceListItem>> = [
+            {
+                data: filteredAndSortedUserWorkspaces,
+                shouldShow: true,
+            },
+        ];
+        return options;
+    }, [filteredAndSortedUserWorkspaces]);
+
+    const headerMessage = filteredAndSortedUserWorkspaces.length === 0 && usersWorkspaces.length ? translate('common.noResultsFound') : '';
```

Comment:
Maybe we could move this condition to a function whose name would better describe what we are checking: 
`filteredAndSortedUserWorkspaces.length === 0 && usersWorkspaces.length`